lib/ansible/modules/network/eos/eos_logging.py

```diff
@@ -0,0 +1,362 @@
+
+    return facility
+
+
+def parse_size(line, dest):
+    size = None
+
+    if dest == 'buffered':
+        match = re.search(r'logging buffered (\S+)', line, re.M)
+        if match:
+            try:
+                int_size = int(match.group(1))
+            except ValueError:
+                int_size = None
+
+            if int_size:
+                if isinstance(int_size, int):
+                    size = str(match.group(1))
+                else:
+                    size = str(10)
+
+    return size
+
+
+def parse_name(line, dest):
+    name = None
+    if dest == 'host':
+        match = re.search(r'logging host (\S+)', line, re.M)
+        if match:
+            name = match.group(1)
+
+    return name
+
+
+def parse_level(line, dest, module):
+    level = None
+
+    if dest is not 'host':
+        match = re.search(r'logging {} (\S+)'.format(dest), line, re.M)
+        if match:
+            if match.group(1) in LEVEL_GROUP:
+                level = match.group(1)
+
+    return level
+
+
+def map_config_to_obj(module):
+    obj = []
+
+    data = get_config(module, flags=['section logging'])
+
+    for line in data.split('\n'):
+        match = re.search(r'logging (\S+)', line, re.M)
+
+        if match.group(1) in DEST_GROUP:
+            dest = match.group(1)
+        else:
+            pass
+
+        obj.append({'dest': dest,
+                    'name': parse_name(line, dest),
+                    'size': parse_size(line, dest),
+                    'facility': parse_facility(line),
+                    'level': parse_level(line, dest, module)})
+
+    return obj
+
+
+def map_params_to_obj(module):
+    obj = []
+
+    if 'aggregate' in module.params and module.params['aggregate']:
+        args = {'dest': '',
+                'name': '',
+                'size': '',
+                'facility': '',
+                'level': '',
+                }
+
+        for c in module.params['aggregate']:
+            d = c.copy()
+
+            for key in args:
+                if key not in d:
+                    d[key] = None
+
+            if d['dest'] != 'host':
+                d['name'] = None
+
+            if 'state' not in d:
+                d['state'] = module.params['state']
+
+            if d['dest'] == 'buffered':
+                if 'size' in d:
+                    d['size'] = str(validate_size(d['size'], module))
+                elif 'size' not in d:
+                    d['size'] = str(10)
+                else:
+                    pass
+
+            if d['dest'] != 'buffered':
+                d['size'] = None
+
+            obj.append(d)
+
+    else:
+        if module.params['dest'] != 'host':
+            module.params['name'] = None
+
+        if module.params['dest'] == 'buffered':
+            if not module.params['size']:
+                module.params['size'] = str(10)
+        else:
+            module.params['size'] = None
+
+        if module.params['size'] is None:
```

Comment:
I think this if-else condition is to handle value of `size`.  Can this be handled outside? This will reduce code duplication